ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs

```diff
@@ -104,7 +115,8 @@ data Mempool m blk idx = Mempool {
       -- they have already been included. (Distinguishing between these two
       -- cases can be done in theory, but it is expensive unless we have an
       -- index of transaction hashes that have been included on the blockchain.)
-      addTxs :: [(GenTxId blk, GenTx blk)] -> m [(GenTx blk, ApplyTxErr blk)]
+      addTxs :: [(GenTxId blk, GenTx blk)]
+             -> m [(GenTx blk, Maybe (ApplyTxErr blk))]
```

Comment:
We have `computeGenTxId :: GenTx blk -> GenTxId blk`, so why are we passing both `GenTx blk` *and* `GenTxId blk`? (What if the given `GenTxId` doesn't match the one returned by `computeGenTxId`?)
Is this just a relic from the time before we had `computeGenTxId` or is there another reason I forgot about?

Reply:
Hm, I don't think there's any good reason for this. I'll make the change.